src/Knp/Component/Pager/Event/Subscriber/Sortable/ArraySubscriber.php

```diff
@@ -2,82 +2,93 @@
 
 namespace Knp\Component\Pager\Event\Subscriber\Sortable;
 
-use \ReflectionClass;
-use Symfony\Component\EventDispatcher\EventSubscriberInterface;
 use Knp\Component\Pager\Event\ItemsEvent;
-use Doctrine\ORM\Query;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\PropertyAccess\PropertyAccess;
+use Symfony\Component\PropertyAccess\PropertyAccessorInterface;
 
 class ArraySubscriber implements EventSubscriberInterface
 {
-
     /**
      * @var string the field used to sort current object array list
      */
-    private $currentSortingFieldGetter;
+    private $currentSortingField;
 
     /**
      * @var string the sorting direction
      */
     private $sortDirection;
 
-    public function items(ItemsEvent $event)
+    /**
+     * @var PropertyAccessorInterface
+     */
+    private $propertyAccessor;
+
+    public function __construct(PropertyAccessorInterface $accessor = null)
     {
-        if (is_array($event->target) && count($event->target) > 1)
```

Comment:
Yep that's it. From my point of view it's better to not restructure the code and in the mean time change the contact(s). But don't worry, I'm not asking you to recreate two separate PRs :)